foldamers/parameters/angle_distributions.py

```diff
@@ -0,0 +1,308 @@
+import os
+import numpy as np
+import matplotlib.pyplot as pyplot
+import mdtraj as md
+from simtk import unit
+from foldamers.cg_model.cgmodel import CGModel
+
+# These functions calculate and plot bond angle and torsion distributions from a CGModel object and pdb trajectory
+
+def calc_bond_angle_distribution(
+    cgmodel,pdbfile,nbins=90,plotfile="angle_hist"
+):
+    """
+    Calculate and plot all bond angle distributions from a CGModel object and pdb trajectory
+
+    :param cgmodel: CGModel() object
+    :type cgmodel: class
+    
+    :param pdbfile: path to pdb trajectory file
+    :type pdbfile: str
+    
+    :param nbins: number of bins spanning the range of 0 to 180 degrees, default = 90
+    :type nbins: int
+    
+    :param plotfile: Base filename for saving bond angle distribution pdf plots
+    :type plotfile: str
+    
+    """
+    
+    # Load in a trajectory pdb file:
+    traj = md.load(pdbfile)
+    nframes = traj.n_frames
+    
+    # Get angle list
+    angle_list = CGModel.get_bond_angle_list(cgmodel)
+    ang_types = [] # List of angle types for each angle in angle_list
+    ang_array = np.zeros((len(angle_list),3))
+    
+    # Angle types:
+    # 1 bb-bb-bb
+    # 2 bb-bb-sc (sc-bb-bb)
+    # 3 bb-sc-sc (sc-sc-bb)
+    # 4 sc-sc-sc
+    
+    # Assign angle types:
+    
+    for i in range(len(angle_list)):
+        ang_array[i,0] = angle_list[i][0]
+        ang_array[i,1] = angle_list[i][1]
+        ang_array[i,2] = angle_list[i][2]
+        at1_type = CGModel.get_particle_type(cgmodel,angle_list[i][0])
+        at2_type = CGModel.get_particle_type(cgmodel,angle_list[i][1])
+        at3_type = CGModel.get_particle_type(cgmodel,angle_list[i][2])
+        
+        if at1_type == 'backbone': # bb-X-X

```

Comment:
I would take a look at things like get_torsion_force_constant for the new types.  You probably want to assemble a combinatorial dictionary of all the possible types (taking into account the fact that a-a-a-b and b-a-a-a are the same type, but with reversed angle).  The iterate the operations over this list.